src/modelinstance.cpp

```diff
@@ -1611,47 +1611,81 @@ Status ModelInstance::inferAsync(const RequestType* requestProto,
+                    OVMS_InferenceResponse* response{nullptr};
+                    CallbackGuard(OVMS_InferenceRequestCompletionCallback_t userCallback, void* userCallbackData, ov::InferRequest& request) :
+                        userCallback(userCallback),
+                        userCallbackData(userCallbackData),
+                        request(request) {}
+                    ~CallbackGuard() {
+                        SPDLOG_DEBUG("Calling user provided callback with success: {}", success);
+                        if (!success) {
+                            userCallback(nullptr, 1, userCallbackData);
+                        } else {
+                            userCallback(response, 0, userCallbackData);
+                        }
+                        SPDLOG_DEBUG("Called user provided callback");
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wterminate"
+                        try {
+                            OV_LOGGER("ov::InferRequest: {} set_callback() with empty lambda", (void*)&request);
+                            request.set_callback([](std::exception_ptr exception_ptr) {});
+                        } catch (std::exception& e) {
+                            SPDLOG_ERROR("Caught critical exception from OpenVINO InferRequest", e.what());
+                            throw e;
+                        } catch (...) {
+                            SPDLOG_ERROR("Caught critical exception from OpenVINO InferRequest");
+                            throw;
+                        }
+#pragma GCC diagnostic pop
+                    }
+                };
+                SPDLOG_DEBUG("Entry of ov::InferRequest callback call");
+                CallbackGuard callbackGuard(userCallback, userCallbackData, inferRequest);
+                if (exception) {
+                    try {
+                        SPDLOG_DEBUG("rethrow_exception");
+                        std::rethrow_exception(exception);
+                    } catch (const std::exception& e) {
+                        SPDLOG_DEBUG("got exception in ov::InferRequest callback: {}", e.what());
+                    } catch (...) {
+                        SPDLOG_DEBUG("got exception in ov::InferRequest callback");
+                        return;
+                    }
+                }
+                std::unique_ptr<ResponseType> res(new ResponseType(this->getName(), this->getVersion()));
+                OutputGetter<ov::InferRequest&> outputGetter(inferRequest);
                 try {
-                    SPDLOG_DEBUG("rethrow_exception");
-                    std::rethrow_exception(exception);
-                } catch (const std::exception& e) {
-                    SPDLOG_DEBUG("got exception in ov::InferRequest callback: {}", e.what());
+                    // TODO created filter based on what is in request, then perform casual serialization for what was NOT in request, and rewrite tensors from request to response for those that were
```

Comment:
what do you mean exactly? do you mean output filtering via output name selection inside request?

Reply:
This is old TODO:
https://github.com/openvinotoolkit/model_server/pull/2801/files/38b759c2a21106cbeab7fa9e77181324047b0b9e#diff-ea20c94d98c26b6ddf560dae96db24b9d293c8a7d75418e744666c52dfa8553cL1642
This idea for slight optimization